lib/mix/lib/mix/utils.ex

```diff
@@ -313,10 +313,7 @@ defmodule Mix.Utils do
         {:ok, _} ->
           File.rm!(target)
           do_symlink_or_copy(source, target, link)
-        {:error, :enoent} ->
-          do_symlink_or_copy(source, target, link)
         {:error, _} ->
-          _ = File.rm_rf!(target)
```

Comment:
What if we do: `_ = not File.dir?(target) && File.rm_rf!(target)`

That's because if we don't have a directory, we should certainly remove it.